### PR TITLE
[EBPF] Minor fixes in KMT tests

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

This PR contains some minor fixes around the KMT local testing process:

* Fix path resolution in packages when symlinks are present
* Clearly show the final result of kmt.test: no test runs/failures/successes
* Fix cross-compilation setup being re-done every time due to the result of the command not being returned 
* Improve information when copying files to remote instances and using the `-e` flag

### CODE REVIEW

File: tasks/kernel_matrix_testing/compiler.py [hunk 1]
```diff
@@ -118,7 +118,7 @@ def exec(self, cmd: str, user="compiler", verbose=True, run_dir: PathOrStr | Non
         self.ensure_running()
 
         # Set FORCE_COLOR=1 so that termcolor works in the container
-        self.ctx.run(
+        return self.ctx.run(
```
Comment (usamasaqib) [inline tasks/kernel_matrix_testing/compiler.py:121]: What does this accomplish?

**Review by usamasaqib — APPROVED** [review]

File: tasks/kernel_matrix_testing/compiler.py (hunk 1 above)
Comment (gjulianm) [reply tasks/kernel_matrix_testing/compiler.py:121]: In [`ensure_ready_for_cross_compile`](https://github.com/DataDog/datadog-agent/blob/main/tasks/kernel_matrix_testing/compiler.py#L196) we check the result of a command ran on the container with `exec()`, but the result was always None as this function was not returning anything, so it was always preparing the cross-compile environment. Adding the return fixes that and now only runs the setup if needed.

**Pull request merged**